tests/adapters_t/batchsystems_t/test_slurm.py

```diff
@@ -0,0 +1,242 @@
+from tests.utilities.utilities import async_return
+from tests.utilities.utilities import run_async
+from tardis.adapters.batchsystems.slurm import SlurmAdapter
+
+from tardis.adapters.batchsystems.slurm import slurm_status_updater
+from tardis.interfaces.batchsystemadapter import MachineStatus
+
+from tardis.exceptions.executorexceptions import CommandExecutionFailure
+
+from functools import partial
+from shlex import quote
+from unittest.mock import patch
+from unittest import TestCase
+
+
+class TestSlurmAdapter(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.mock_config_patcher = patch(
+            "tardis.adapters.batchsystems.slurm.Configuration"
+        )
+        cls.mock_async_run_command_patcher = patch(
+            "tardis.adapters.batchsystems.slurm.async_run_command"
+        )
+        cls.mock_config = cls.mock_config_patcher.start()
+        cls.mock_async_run_command = cls.mock_async_run_command_patcher.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.mock_config_patcher.stop()
+        cls.mock_async_run_command_patcher.stop()
+
+    def setUp(self):
+        self.cpu_ratio = 0.5
+        self.memory_ratio = 0.25
+
+        self.command = 'sinfo --Format="statelong,cpusstate,allocmem,memory,features,nodehost" -e --noheader -r --partition test_part'
+        #  self.command = 'sinfo --Format="statelong,cpusstate,allocmem,memory,features,nodehost" -e --noheader --partition test_part'
+
+        self.command_wo_options = 'sinfo --Format="statelong,cpusstate,allocmem,memory,features,nodehost" -e --noheader -r'
+        #  self.command_wo_options = 'sinfo --Format="statelong,cpusstate,allocmem,memory,features,nodehost" -e --noheader'
+
+        return_value = "\n".join(
+            [
+                f"mixed      2/2/0/4   6000    24000   VM-1   host-10-18-1-1",
+                f"mixed      3/1/0/4   15853   22011   VM-2   host-10-18-1-2",
+                f"mixed      1/3/0/4   18268   22011   VM-3   host-10-18-1-4",
+                f"mixed      3/1/0/4   17803   22011   VM-4   host-10-18-1-7",
+                f"draining   0/4/0/4   17803   22011   draining_machine   draining_machine",
+                f"idle       0/4/0/4   17803   22011   idle_machine   idle_machine",
+                f"drained    0/4/0/4   17803   22011   drained_machine   drained_machine",
+                f"powerup    0/4/0/4   17803   22011   power_up_machine   power_up_machine",
+            ]
+        )
+
+        self.mock_async_run_command.return_value = async_return(
+            return_value=return_value
+        )
+
+        self.setup_config_mock(options={"partition": "test_part"})
+
+        self.slurm_adapter = SlurmAdapter()
+
+    def tearDown(self):
+        self.mock_async_run_command.reset_mock()
+
+    def setup_config_mock(self, options=None):
+        self.config = self.mock_config.return_value
+        self.config.BatchSystem.ratios = {
```

Comment:
According to your adapter code, Slurm does not support that type of equations, why are they set in the configuration?